test/function/samples/preserve-for-of-iterable/iterables.js

```diff
@@ -0,0 +1,18 @@
+export const zeroToFive = {
+    [Symbol.iterator]() {
+        return {
+            current: 0,
+            last: 5,
+            next() {
+                const ret = this.current < this.last
+                    ? { done: false, value: this.current++ }
+                    : { done: true };
+                
+                // assert later
+                global.foo = this.current;
```

Comment:
A cooler way than mutating a global and testing our test isolation would be to export a mutable `export let ...` binding and checking that.